### ISSUE TITLE
[Feature Request] Blocks for detecting and controlling features

### ISSUE THREAD

**Issue by just-a-hriday** [TurboWarp/scratch-vm#55]
It would be great if there were blocks that controlled and detected TurboWarp features like 60 FPS mode, Warp Timer, Custom Screen Size, High Quality Pen etc.

**Comment by GarboMuffin** [comment]
I don't want to add new custom blocks, but I still want to find a way to do this, so I was thinking of storing some internal data in a [script comment in the stage](https://user-images.githubusercontent.com/33787854/118027170-3e954b00-b327-11eb-8215-bf5b9cff3ee6.png). This process could be automated by the GUI. This would allow you to configure your projects to run at 60 FPS by default or anything in the advanced settings menu. One thing that this would not allow is changing options at runtime, but I don't think many projects want to do that anyways.

Detecting things like 60 FPS mode and custom screen size is already possible: https://scratch.mit.edu/projects/487694716/, https://scratch.mit.edu/projects/475562469/

I don't think high quality pen and especially warp timer need detectors. Projects will either use it or they won't.

Thoughts? I do already have a prototype for this

**Comment by RokCoder** [comment]
TLDR: I think it's a very good approach. 

It's not the most syntactically memorable method _but_ it doesn't need to be if it's being automated. The comment idea has the big bonus of being something that you literally can't change during runtime. I don't see any benefit to being able to alter these parameters at runtime, either. It had crossed my mind that simply using global variables would work (TWFrameRate=50 for example) but I think comments are better for their const-ness.

I haven't had much luck with detecting frame rate accurately using Scratch scripting. I had originally included code to detect frame rate, infinite clones and limitless (for the non-yielding sfx) in my Home Computer emulator. Infinite clones and limitless are straightforward but I had a handful of comments complaining that it wasn't allowing the project to run even in TurboWarp due to the checks. I was giving the framerate check quite a leeway, too. Once I removed the framerate check, the complaints stopped. It becomes a moot point if it's able to be set via the automated comment approach anyway.

**Comment by GarboMuffin** [comment]
There's a prototype of this on https://staging.turbowarp.org/ that adds a button at the bottom of the advanced settings menu to store settings in a stage comment. Custom stage size and warp timer won't be saved automatically. Feel free to start using it and letting me know if something's completely broken.

One unfortunante thing is that users will have to save the project from TurboWarp and upload it to Scratch or figure out that they just have to copy a cryptic comment. I don't like that, but I don't think there's much that can be done about it as long as the data is stored in the project JSON.

Will probably make it into the main website by ~Monday

**Comment by just-a-hriday** [comment]
Is it a regular scratch comment in the project or is it a comment in the JSON file? Cause I don't see any regular scratch comments 🤔

**Comment by GarboMuffin** [comment]
It's a regular Scratch comment in the stage. Remember that you have to manually go to https://staging.turbowarp.org/ and press the button at the end of the advanced settings modal.

**Comment by just-a-hriday** [comment]
I did that, but there was still no comment.

**Comment by RokCoder** [comment]
It's working fine here

**Comment by just-a-hriday** [comment]
What OS and browser are you using? I'm on Microsoft Edge Beta 91.0.864.27 in Windows 10 20H2.

**Comment by RokCoder** [comment]
> What OS and browser are you using? I'm on Microsoft Edge Beta 91.0.864.27 in Windows 10 20H2.

I'm using Chrome (90.0.4430.212) in the same Windows as you but have checked Edge (90.0.818.62) and it's working fine there, too.

**Comment by GarboMuffin** [comment]
This feature went into production a couple days ago. I'm keeping this open for a bit longer as the current solution is not finalized and lacks certain features (you will be notified here if any backwards incompatible changes are made)

**Comment by just-a-hriday** [comment]
> I'm using Chrome (90.0.4430.212) in the same Windows as you but have checked Edge (90.0.818.62) and it's working fine there, too.

Hmm... I'll try again.`
